### PR TITLE
YAML grammar files

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request partially fixes the issue at #3070.

The scopes for `vendor/grammars/NSIS` and `vendor/grammars/awk-sublime` were not recognized because the Sublime Text XML files were recently removed. The authors only kept the YAML files from which the XML files are generated.

Thus, this pull request adds support for YAML syntax definitions in `script/convert-grammars`.
A few new scopes were detected via `.YAML-tmLanguage` files. I added them in `grammars.yml`.

:warning: Of course, for this to work, GitHub backend for syntax highlighting must also recognize YAML grammar files.

### CODE REVIEW

File: script/convert-grammars [hunk 1]
```diff
@@ -42,7 +42,7 @@ class DirectoryPackage
       case File.extname(path.downcase)
       when '.plist'
         path.split('/')[-2] == 'Syntaxes'
-      when '.tmlanguage'
+      when '.tmlanguage', '.yaml-tmlanguage', '.sublime-syntax'
```
Comment (pchaigno) [inline script/convert-grammars]: Sublime Text YAML syntax definitions use the `.YAML-tmLanguage` file extension, and more recently, `.sublime-syntax`.

File: script/convert-grammars [hunk 2]
```diff
@@ -148,6 +148,17 @@ def load_grammar(path)
...
+  when '.yaml-tmlanguage', '.sublime-syntax'
+    content = File.read(path)
+    # Attempt to parse YAML file even if it has a YAML 1.2 header
+    if content.lines[0] =~ /^%YAML[ :]1\.2/
+      content = content.lines[1..-1].join
+    end
+    begin
+      YAML.load(content)
+    rescue Psych::SyntaxError => e
+      $stderr.puts "Failed to parse YAML grammar '#{path}'"
```
Comment (pchaigno) [inline script/convert-grammars]: Most syntax files declare a YAML 1.2 syntax (`%YAML 1.2`) although they are YAML 1.1 compatible.
Here, the YAML version header is stripped off before parsing. If parsing fails, it displays a warning.

File: script/convert-grammars [hunk 3]
```diff
@@ -185,7 +196,7 @@ def load_grammars(tmp_dir, source, all_scopes)
 
   p.fetch(tmp_dir).map do |path|
     grammar = load_grammar(path)
-    scope = grammar['scopeName']
+    scope = grammar['scopeName'] || grammar['scope']
```
Comment (pchaigno) [inline script/convert-grammars]: In YAML grammar files, the scope is under the `scope` key -- as opposed to the usual `scopeName` key.

**Comment by arfon** [comment]
/ cc @aroben incase he has a chance to give this a quick look.

**Comment by aroben** [comment]
> Of course, for this to work, GitHub backend for syntax highlighting must also recognize YAML grammar files.

GitHub uses the JSON files this script creates rather than parsing the original files directly. So if this script can turn YAML into JSON then it should work fine on GitHub.

**Comment by aroben** [comment]
This seems reasonable to me 👍

**Comment by arfon** [comment]
> @arfon Do you want to wait for a specific version to merge this or should I go ahead?
 Show all checks

Nah, let's get this merged in.

**Pull request merged**